constraints/miden-vm/range_checker.air

```diff
@@ -0,0 +1,61 @@
+mod RangeCheckerAir
+
+### Helper functions ##############################################################################
+
+# Returns binary negation of the value.
+fn binary_not(e: scalar) -> scalar:
+    return 1 - e
+
+# Returns array of mutually exclusive multiplicity flags.
+# f[0] set to 1 when we don't include the value into the running product.
+# f[1] set to 1 when we include the value into the running product.
+# f[2] set to 1 when we include two copies of the value into the running product.
+# f[3] set to 1 when we include four copies of the value into the running product.
+fn get_multiplicity_flags(s0: scalar, s1: scalar) -> vector[4]:
+    return [!s0 & !s1, s0 & !s1, !s0 & s1, s0 & s1]
+
+
+### Helper evaluators #############################################################################
+
+# Enforces that column must be binary.
+ev is_binary([v]):
+    enf v^2 = v
+
+# Enforces correct transition from 8-bit to 16-bit section of the table.
+ev transition_8_to_16_bit_validity([t, v]):
+    # Ensure that values in column t can flip from 0 to 1 only once
+    enf t * binary_not(t') = 0
+
+    # Ensure that when column t flips, column v must equal 255
+    enf v = 255 when t' & !t
+
+    # Ensure that when column t flips, v' must be reset to 0
+    enf v' = 0 when t' & !t
+
+# Enforces the internal correctness of the 16-bit section of the Range Checker.
+ev range_checker_8bit([t, s0, s1, v], [p0]):
```

Comment:
nit: Let's rename this evaluator to "virtual_table" and update the comment to "The virtual table enforces an 8-bit range check for each row transition in the 16-bit section of the range checker, which enforces its internal correctness."